core/server/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -510,6 +512,12 @@ public void addInodeFromJournal(JournalEntry entry) {
 
       if (directory.getName().equals(ROOT_INODE_NAME)) {
         // This is the root inode. Clear all the state, and set the root.
+        if (SecurityUtils.isSecurityEnabled(MasterContext.getConf())
+            && mRoot != null && !mRoot.getUserName().equals(directory.getUserName())) {
+          // user is not the owner of journal root entry
+          throw new IOException(
```

Comment:
Right, good idea. Thanks! I missed the similar handlings in `processJournalEntry`.